Sources/Build/YAML.swift

```diff
@@ -8,56 +8,37 @@
  See http://swift.org/CONTRIBUTORS.txt for Swift project authors
 */
 
-import struct Utility.Path
-import struct libc.FILE
-import func libc.fclose
-import POSIX
-
-class YAML {
-    let path: String
-    private let fp: UnsafeMutablePointer<FILE>
-
-    init(path: String...) throws {
-        self.path = Path.join(path)
-        fp = try fopen(self.path, mode: .Write)
-    }
+protocol YAMLRepresentable {
+    var YAML: String { get }
+}
 
-    func close() {
-        fclose(fp)
+extension String: YAMLRepresentable {
+    var YAML: String {
+        if self == "" { return "\"\"" }
+        return self
     }
+}
 
-    func write(anys: Any...) throws {
-        var anys = anys
-        try fputs(anys.removeFirst() as! String, fp)
-        if !anys.isEmpty {
-            try fputs(anys.map(toYAML).joinWithSeparator(""), fp)
-        }
-        try fputs("\n", fp)
-
+extension Bool: YAMLRepresentable {
+    var YAML: String {
+        if self { return "true" }
+        return "false"
     }
 }
 
-private func toYAML(any: Any) -> String {
-
-    func quote(input: String) -> String {
-        for c in input.characters {
-            if c == "@" || c == " " || c == "-" {
-                return "\"\(input)\""
+extension Array: YAMLRepresentable {
+    var YAML: String {
+        let stringArray = self.flatMap { $0 as? String }
+        guard stringArray.count == self.count else { return "" }
```

Comment:
I thought about it when I was doing it but as you said Array will not conform to `YAMLRepresentable` but it will work for the current use case and provide better compile time checks. Updating code.